constraint_test.go

```diff
@@ -62,6 +62,14 @@ func TestConstraintCheck(t *testing.T) {
 		{"~> 1.0.9.5", "1.0.9.6", true},
 		{"~> 1.0.9.5", "1.0.9.5.0", true},
 		{"~> 1.0.9.5", "1.0.9.5.1", true},
+		{">=2.1.0-a", "2.1.1-beta", true},
```

Comment:
I'm not sure what rubygems does but my experience with other tools is that there is a further constraint here: if I request `2.1.0-a` then that can match any prerelease with the base `2.1.0`, but would _not_ match a prerelease with another base such as `2.1.1-beta`.

You can see this in [npm's constraint syntax](https://docs.npmjs.com/misc/semver#prerelease-tags), for example.

It feels "right" to me that specifying `>= 2.1.0-a` would _not_ match `2.1.1-beta` for reasons similar to what is stated on that `npm` page, but that's definitely subjective and I'm certainly bringing my biases from other software here. If what you implemented here is rubygems behavior then I'd be fine with moving forward with it specified this way on _that_ basis, since being consistent with something is better than going our own way.

Reply:
@apparentlymart,
Thanks! I think that is a good idea, and agree that it's a saner behavior.

Reply:
Good catch.